src/plugins/analysis/device_tree/internal/device_tree_utils.py

```diff
@@ -72,13 +72,15 @@ def header_has_illegal_values(header: DeviceTreeHeader, max_size: int) -> bool:
         header.struct_block_size,
         header.strings_block_size,
     ]
-    return header.version > 20 or any(n > max_size or n > header.size for n in values) or header.size > max_size
+    return (
+        header.version > 20  # noqa: PLR2004
+        or any(n > max_size or n > header.size for n in values)
+        or header.size > max_size  # noqa: PLR2004, RUF100
+    )  # noqa: PLR2004, RUF100
```

Comment:
...
black vs ruff 
1. ruff adds "# noqa: PLR2004"
2. black breaks the line
3. the "PLR2004" doesn't fit anymore so ruff adds a "RUF100" and a new "PLR2004"
4. the line is now even longer so black breaks it again
5. the second "PLR2004" doesn't fit anymore so ruff adds another "RUF100" and also a new "PLR2004"